### PR TITLE
HPCC4J-471 Upgrade to ANTLR 4.10.1

### PR DESCRIPTION
**Pull request by rpastrana**
- Bumps ANTLR dep to 4.10.1
- Postpends gen on generated parser package
- Includes *.interp files

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [ ] I have created a corresponding JIRA ticket for this submission
- [ ] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by rpastrana** [comment]
@drealeed please review. Also, it appears all related junits are currently bypassed due to absence of expected test files. Please provide your original test resources. thanks.

**Comment by drealeed** [comment]
I won't have time this afternoon but tomorrow will pull it down and figure out how to re-enable and run the tests for it.

**Comment by leonarta** [comment]
@rpastrana I'd suggest expanding the top-level `.gitignore` file. I was looking at this locally and as it stands now IJ would have been encouraging me to check in several ANTLR-generated files (which weren't generated in earlier versions), and a bunch of files related to IntelliJ itself.

The following incorporates the "standard" template for Java, plus the others files we need:

````
# --------------------------------------------------------------------------------------------
# .gitignore - Specifies files to exclude from git commits
# --------------------------------------------------------------------------------------------

# ANTLR-generated files (which we build on the fly via antlr4-maven-plugin)
EclRecord*.*
!EclRecord.g4
!EclRecordReader.java

# IntelliJ files
*.iml
**/.idea/

# Eclipse files
**/.gitattributes
**/.gitconfig
**/.project
**/.classpath
**/.cproject
**/.settings

# Files generated by your unit tests belongs in a target directory!
**/target/

# --------------------------------------------------------------------------------------------
# The rest is derived from https://github.com/github/gitignore/blob/main/Java.gitignore
# --------------------------------------------------------------------------------------------

# Compiled class file
*.class

# Log file
*.log

# BlueJ files
*.ctxt

# Mobile Tools for Java (J2ME)
.mtj.tmp/

# Package Files #
*.jar
*.war
*.nar
*.ear
*.zip
*.tar.gz
*.rar

# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
hs_err_pid*
replay_pid*
````

**Comment by leonarta** [comment]
Alternately, the EclRecord lines could be in `wsclient/.gitignore` rather than the top-level file.  That might be better if you're committed to having a multilevel definition (though it really doesn't seem necessary).

File: wsclient/src/main/java/org/hpccsystems/ws/client/antlr/gen/EclRecord.interp [hunk 1]
```diff
@@ -0,0 +1,109 @@
+token literal names:
```
Comment (leonarta) [inline wsclient/src/main/java/org/hpccsystems/ws/client/antlr/gen/EclRecord.interp:1]: I think this file should probably not be committed, and falls under my comment about expanding the .gitignore file

File: wsclient/src/main/java/org/hpccsystems/ws/client/antlr/gen/EclRecordLexer.interp [hunk 2]
```diff
@@ -0,0 +1,117 @@
+token literal names:
```
Comment (leonarta) [inline wsclient/src/main/java/org/hpccsystems/ws/client/antlr/gen/EclRecordLexer.interp:1]: I think this file should probably not be committed, and falls under my comment about expanding the .gitignore file

File: wsclient/pom.xml [hunk 3]
```diff
@@ -90,10 +90,10 @@
             </goals>
             <configuration>
               <sourceDirectory>${project.basedir}/src/main/java/org/hpccsystems/ws/client/antlr</sourceDirectory>
-              <outputDirectory>${project.basedir}/src/main/java/org/hpccsystems/ws/client/antlr</outputDirectory>
+              <outputDirectory>${project.basedir}/src/main/java/org/hpccsystems/ws/client/antlr/gen</outputDirectory>
```
Comment (leonarta) [inline wsclient/pom.xml:93]: What's the rationale behind shifting this down a level?

**Pull request closed without merging**